Content.Server/Body/Systems/MetabolizerSystem.cs

```diff
@@ -138,6 +139,9 @@ private void TryMetabolize(Entity<MetabolizerComponent, OrganComponent?, Solutio
             var list = solution.Contents.ToArray();
             _random.Shuffle(list);
 
+            TryComp<BloodstreamComponent>(solutionEntityUid.Value, out var bloodstream);
```

Comment:
Event should be raised here. Organs and such use Metabolizer and don't have bloodstreams. For example: Lungs, which metabolize as lungs regardless of bloodstream. 

```suggestion
            var ev = new MetabolismExclusionEvent(list);
            RaiseLocalEvent(solutionEntityUid.Value, ref ev);
            
            _random.Shuffle(ev.list);
```

If you need me to I can push changes to get it working.